### PR TITLE
build system: rewrites

### PR DESCRIPTION
**Pull request by jchavarri**
Applies @anmonteiro's suggestions from https://github.com/melange-re/documentation-site/pull/5.

### CODE REVIEW

File: docs/build-system.md [hunk 1]
```diff
@@ -189,19 +187,78 @@ $ node _build/default/app/app.js
...
 
-One thing to note is that we have to look for the `app.js` file inside an `app`
-folder, but we don’t have any such folder in our sources. The reason why this
-folder is created is to support multiple `melange.emit` stanzas in the same
-folder. To support this scenario, Dune will use the `target` field defined in
-the `melange.emit` to place the artifacts generated from a `melange.emit` stanza
-in the following folder:
+#### JavaScript artifacts layout
+
+One thing to note is that in the command above we had to look for the `app.js`
```
Comment (anmonteiro) [inline docs/build-system.md:192]: > One thing to note is that

remove, not adding to the sentence

File: docs/build-system.md [hunk 2]
```diff
@@ -189,19 +187,78 @@ $ node _build/default/app/app.js
...
+```
+
+Then, the JavaScript artifacts for `foo.ml` will be placed under:
+
+```text
+_build/default/emit/app/lib/foo.js
+```
+
+More generically:
+
```
Comment (anmonteiro) [inline docs/build-system.md:233]: it's really great to have this documented, so we can point people here!

File: docs/build-system.md [hunk 3]
```diff
@@ -189,19 +187,78 @@ $ node _build/default/app/app.js
...
+The path to the generated JavaScript file from `$name.ml` will be:
+
+```text
+_build/default/$melange-emit-folder/$target/$path-to-source-file/$name.js
+```
+
+#### Guidelines for `melange.emit`
+
+Here are some recommendations about the `melange.emit` stanza that can help with
+the maintenance and performance of Melange projects:
```
Comment (anmonteiro) [inline docs/build-system.md:249]:
```suggestion
The following recommendations around `melange.emit` have been tested within
large industrial projects, and have proven to be helpful guidelines to deal with
complexity, maintenance and build performance.
```

File: docs/build-system.md [hunk 4]
```diff
@@ -189,19 +187,78 @@ $ node _build/default/app/app.js
...
+
+- To simplify access to the generated JavaScript files from tools like Webpack,
+  it is recommended to place the `dune` files containing the `melange.emit`
+  stanzas in the project’s root folder. This arrangement ensures that the
```
Comment (anmonteiro) [inline docs/build-system.md:253]:
```suggestion
  stanzas in the project’s root folder. This ensures that the
```

File: docs/build-system.md (hunk 1 above)
Comment (jchavarri) [reply docs/build-system.md]: https://github.com/melange-re/documentation-site/pull/25/commits/3aaaa01f303e8217b4eb543966b527d3b6fd2fbf

**Pull request merged**